include/boost/url/detail/impl/remove_dot_segments.ipp

```diff
@@ -0,0 +1,518 @@
+        // prefix of "/./" or "/.", where "." is
+        // a complete path segment, then replace
+        // that prefix with "/" in the input
+        // buffer; otherwise,
+        if (s.starts_with("/./"))
+        {
+            s.remove_prefix(2);
+            continue;
+        }
+        if (s == "/.")
+        {
+            // equivalent to replacing s with '/'
+            // and executing the next iteration
+            append(s.substr(0, 1));
+            s.remove_prefix(2);
+            continue;
+        }
+
+        // C. if the input buffer begins with a
+        // prefix of "/../" or "/..", where ".."
+        // is a complete path segment, then
+        // replace that prefix with "/" in the
+        // input buffer and remove the last
+        // segment and its preceding "/"
+        // (if any) from the output buffer;
+        // otherwise,
+        if (s.starts_with("/../"))
+        {
+            std::size_t p = find_last_slash();
+            if (p != string_view::npos)
+            {
+                // "erase" [p, end]
+                dest = dest0 + p;
+            }
+            else if (!remove_unmatched)
+            {
+                append(s.substr(0, 3));
+            }
+            s.remove_prefix(3);
+            continue;
+        }
+        if (s == "/..")
+        {
+            std::size_t p = find_last_slash();
+            if (p != string_view::npos)
+            {
+                // erase [p, end]
+                dest = dest0 + p;
+                // equivalent to replacing s with '/'
+                // and executing the next iteration.
+                // this is the only point that would
+                // require input memory allocations
+                // in remove_dot_segments
+                append(s.substr(0, 1));
+            }
+            else if (remove_unmatched)
+            {
+                append(s.substr(0, 1));
+            }
+            else if (!remove_unmatched)
+            {
+                append(s.substr(0, 3));
+            }
+            s.remove_prefix(3);
+            continue;
+        }
+
+        // E.  move the first path segment in the
+        // input buffer to the end of the output
+        // buffer, including the initial "/"
+        // character (if any) and any subsequent
+        // characters up to, but not including,
+        // the next "/" character or the end of
+        // the input buffer.
+        std::size_t p = s.find_first_of('/', 1);
+        if (p != string_view::npos)
+        {
+            append(s.substr(0, p));
+            s.remove_prefix(p);
+        }
+        else
+        {
+            append(s);
+            s = {};
+        }
+    }
+
+    // 3. Finally, the output buffer is returned
+    // as the result of remove_dot_segments.
+    return dest - dest0;
+}
+
+int
+normalized_path_compare(
+    string_view s0_init,
+    string_view s1_init,
+    bool s0_remove_unmatched,
+    bool s1_remove_unmatched) noexcept
```

Comment:
What do these bools do?

Reply:
They remove unmatched `..`s, like RFC requires for absolute paths or keep the unmatched `..`, like we expect for relative references.